### PR TITLE
New feature: Make invitation/reminder sending rate configurable

### PR DESCRIPTION
**Pull request by ViliusS**
Dev: This patch makes currently hardcoded 20 second automatic email sending rate configurable.

Dev: Most SMTP providers limit sending rates on their side. This patch, combined with "Email batch size" parameter, now allows for a very flexible solution to the problem. For example, you can set LimeSurvey to send 30 emails per 1 minute OR you can set it to 1 email per 2 seconds which could be a big improvement on some SMTP platforms to not being blocked by spam filters.

Dev: While at it, convert related strings to be translatable, allow for plural translation forms and make notice to the survey administrator more sensical when a "Cancel automatic sending" button is clicked.

With small adjustments in _email.php (`<span> -> <p>`) the patch could be merged to LS3.x branch too.

P.S. This patch is against master because https://github.com/LimeSurvey/LimeSurvey/blob/master/CONTRIBUTING.md says it should be so, but only **after** I pulled a request github PR template said it should be in develop? 

P.P.S. I could not create a ticket on bugs.limesurvey.org because I'm using my github account to login to limesurvey.org, but it seems Mantis doesn't support this?

### CODE REVIEW

**Comment by Shnoulle** [comment]
I like it !

File: application/views/admin/globalsettings/_email.php [hunk 1]
```diff
@@ -93,6 +93,13 @@
                 <input class="form-control"  type='text' size='5' id='maxemails' name='maxemails' value="<?php echo htmlspecialchars(getGlobalSetting('maxemails')); ?>" />
             </div>
         </div>
+        <div class="form-group">
+            <label class="control-label" for="sendingrate"><?php eT("Email sending rate:"); ?></label>
+            <div>
+                <?php echo Chtml::textField("sendingrate", getGlobalSetting('maxemails'), array('class' => 'form-control', 'size' => '5')); ?>
```
Comment (Shnoulle) [inline application/views/admin/globalsettings/_email.php:99]: numberField maybe ? 

Else : https://github.com/LimeSurvey/LimeSurvey/blob/11d333b86731acb9be0b53d9f7b629a36a65c7f8/application/helpers/globalsettings_helper.php#L20 :)

Comment (ViliusS) [reply application/views/admin/globalsettings/_email.php]: I tried to make a patch to be compatible with LS3.x branch as close as possible. I cannot find information when App()->getConfig($settingname)  was deprecated. Can I use it in LS3?

Comment (ViliusS) [reply application/views/admin/globalsettings/_email.php]: I should I use App()->getConfig() or Yii::app->getConfig() in templates?

Comment (Shnoulle) [reply application/views/admin/globalsettings/_email.php]: This is not really fixed but some of dev prefer to use `App()`

Comment (Shnoulle) [reply application/views/admin/globalsettings/_email.php]: > I tried to make a patch to be compatible with LS3.x branch as close as possible. I cannot find information when getGlobalSettings() was deprecated. Can I use App()->getConfig() in LS3?

Yes :) https://github.com/LimeSurvey/LimeSurvey/blob/e8237925b5915b739c4049183d171b714b308585/application/helpers/globalsettings_helper.php#L58

Comment (ViliusS) [reply application/views/admin/globalsettings/_email.php]: Done.

**Comment by Shnoulle** [comment]
I make some test and review it again :)

**Comment by Shnoulle** [comment]
Did you kow if you can move to develop branch ? Seems new feature must go to develop branch ([see my pull request](https://github.com/LimeSurvey/LimeSurvey/pull/1632))

**Comment by ViliusS** [comment]
It should be compatible with develop branch however I don't know how to do it now? Should I create a new pull request for develop branch or can someone change it from your side?

P.S. CONTRIBUTING.md and other documentations really needs to be updated, because now there are mismatched all over the place. I'm still not sure which branch I need to use.

**Comment by Shnoulle** [comment]
> It should be compatible with develop branch however I don't know how to transfer to it now. Should I create a new pull request for develop branch or can someone change it from your side?
> 
> P.S. CONTRIBUTING.md and other documentations really needs to be updated, because now there are mismatches all over the place. I'm still not sure which branch I need to use.

Yes … sure … i still fix my pull request … https://github.com/LimeSurvey/LimeSurvey/commit/b4f74e805722559efc62c6549c4251a535b11c8f

Unsure : personaly i merge locally develop … ans start fixing …

**Comment by ViliusS** [comment]
The easiest way I see now is that someone with write access to the repository change the base https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by ViliusS** [comment]
Should be OK now.

**Comment by Shnoulle** [comment]
Really like this !!
Mail one by one each second :)

Peek 03-11-2020 16-41

**Comment by Shnoulle** [comment]
> Should be OK now.

Thanks a lot !

**Comment by ViliusS** [comment]
Could you merge this to LS3.x too? I know that there is a consensus to not add new features to LS3.x anymore, but since LS4 is still not stable and this is more a change to existing feature than a completely new feature, it would be great to have it in LS3.x We are still using it extensively.

**Comment by Shnoulle** [comment]
Not me … @cdorin93 can answer

**Comment by ViliusS** [comment]
@c-schmitz , @cdorin93 any chance to have this merged to LS3.x? I would really hate to have to manage this patch on my side for such simple and useful change until LS4 becomes stable.

**Comment by cdorin93** [comment]
Hello @ViliusS , and thank you for the nice feature.
I talked to my colleagues and we will have it merged into develop/LS4.
We hope to bring LS4 to stable as soon as possible - 4.4 will be a an important release.

**Comment by maziminke** [comment]
@cdorin93 Since it will surely take quite some time until LS 4.x is really stable, can you please consider merging this into LS3.x as well? Many, many users will benefit from this!

@ViliusS Thanks a lot for this great contribution!

**Pull request merged**